google/cloud/datastore/query.py

```diff
@@ -593,8 +593,12 @@ def _next_page(self):
             # skipped all of the results yet. Don't return any results.
             # Instead, rerun query, adjusting offsets. Datastore doesn't process
             # more than 1000 skipped results in a query.
+            old_query_pb = query_pb
+            query_pb = query_pb2.Query()
+            query_pb._pb.CopyFrom(old_query_pb._pb)  # copy for testability
```

Comment:
@craiglabenz is this going to hit the bad path for perf and protoplus?

I have hesitation about making a product change to make it possible to test. Can you shed more light on this change @tseaver

Reply:
For perf:
- We are dealing with the raw protos here -- no protoplus attributes are touched, and `CopyFrom` is super-fast (implemented in C).
- The copy will only be made once per page, in the edge case where we are skipping over more than a page of results.
For correctness:
- Scribbling on the original `query_pb` is a puzzlement, and might actually defeat some kind of retry.